App Delegate/MITSlidingViewController.m

```diff
@@ -448,16 +453,15 @@ - (void)_showModuleWithModuleItem:(MITModuleItem*)moduleItem animated:(BOOL)anim
 {
     NSParameterAssert(moduleItem);
 
-    // Call the application delegate directly to change the module so we follow the same
-    // event handling as everything else. This is only needed at this point because this class
-    // only deals with view controllers, while the actual module contents exist outside of the
-    // view controller lifecycle. This needs to be reworked.
-    // (bskinner - 2014.11.07, updated 2014.12.01)
     UIViewController *moduleViewController = [self _moduleViewControllerWithName:moduleItem.name];
     NSAssert(moduleViewController,@"module with name %@ does not have an associated view controller.",moduleItem.name);
 
-
-    [[MIT_MobileAppDelegate applicationDelegate] showModuleWithTag:moduleItem.name animated:animated];
+    [self _willHideTopViewController:self.visibleViewController];
+    __weak MITSlidingViewController *weakSelf = self;
+    [self setVisibleViewController:moduleViewController animated:animated completion:^{
+        MITSlidingViewController *blockSelf = weakSelf;
+        [blockSelf _didShowTopViewController:blockSelf.visibleViewController];
```

Comment:
No need. If _blockSelf_ is nil here, the entire method call is a NOOP and there's no other operations being performed.